backend/app/services/implementations/onboarding_request_service.py

```diff
@@ -38,3 +39,57 @@ def create_onboarding_request(self, userInfo):
             raise e
 
         return new_onboarding_request.to_serializable_dict()
+
+    def get_all_onboarding_requests(self, number=5, offset=0, role="", status=""):
+        onboarding_request_dtos = []
+
+        try:
+            filteredRequests = OnboardingRequest.objects()
+            if role:
+                filteredRequests = filteredRequests.filter(info__role=role)
+            if status:
+                filteredRequests = filteredRequests.filter(status=status)
+            for request in filteredRequests:
```

Comment:
To get the efficiency benefits you should probably do the limiting here instead of below:
```suggestion
            for request in filteredRequests.skip(offset).limit(number or 0):
```
(The above snippet also assumes you make `number` default to `None`.)